### PR TITLE
S1T4- Loop function built and unit tested

### PR DESCRIPTION
**Pull request by j-add**

### CODE REVIEW

File: index.php [hunk 1]
```diff
@@ -5,6 +5,4 @@
 $db = getDB();
 $data = getData($db);
```
Comment (Parsley-Afroman) [inline index.php:6]: Is it possible, as you had suggested to me, that, by assigning the variables here, a user could reassign the variable; therefore, is it possible to pass these to index as purely functions or already set variables?

File: functions.php [hunk 2]
```diff
@@ -18,4 +18,43 @@ function getData(object $db): array {
...
+        if ($record['albumImage'] == NULL || $record['albumImage'] == "") {
+            $record['albumImage'] = $placeholderImage;
+        }
+        // Output each data field
+        $albumImage = '<div><img class="albumImg" src="' . $record['albumImage'] . '" alt="Album cover for ' . $record['albumName'] . ' by ' . $record['artistName'] . '" /></div>';
+        $albumName = '<h1 class="albumName">' . $record['albumName'] . '</h1>';
+        $artistName = '<h2 class="artistName">' . $record['artistName'] . '</h2>';
+        $genre = '<div><p class="genre ' . $record['genre'] . '">' . $record['genre'] . '</p></div>';
+        //If purchaseDate is NULL do not include it in output string
+        if ($record['purchaseDate'] != NULL) {
```
Comment (Parsley-Afroman) [inline functions.php:46]: Probably need to add something in here to ensure that no albums can be submitted with a future date.

**Review by Parsley-Afroman — COMMENTED** [review]
There are a couple things to consider

File: functions.php (hunk 2 above)
Comment (j-add) [reply functions.php:46]: Good point. Not too concerned about this at the moment, as no data is being submitted, but will keep it in mind for Story 2, when I am validating user input.

File: index.php (hunk 1 above)
Comment (j-add) [reply index.php:6]: I could maybe set them as constants but probably shouldn't bundle functions together.
Want to keep my functions file just for setting functions.

Do you have any suggestions?

Comment (Parsley-Afroman) [reply index.php:6]: I have had a look and a suggestion is to put them in a class and declare them as private, and instead using getters for attaining the information.

File: functions.php (hunk 2 above)
Comment (Parsley-Afroman) [reply functions.php:46]: Okay, makes sense; giving yourself a small Technical Debt

**Review by Parsley-Afroman — APPROVED** [review]
I am happy that, from your responses, that the issues I have raised are already within your attention/scope of work to do.

The function aspect of my commentary is one of personal choice but it may be worth thinking of how you can keep variables from being reassigned.

**Pull request merged**